src/bootstrap/setup.rs

```diff
@@ -86,3 +92,44 @@ d) Install Rust from source"
     };
     Ok(template.to_owned())
 }
+
+// install a git hook to automatically run tidy --bless, if they want
+fn install_git_hook_maybe(src_path: &Path) -> io::Result<()> {
+    let mut input = String::new();
+    println!(
+        "Rust's CI will automatically fail if it doesn't pass `tidy`, the internal tool for ensuring code quality.
+If you'd like, x.py can install a git hook for you that will automatically run `tidy --bless` on each commit
+to ensure your code is up to par. If you decide later that this behavior is undesirable,
+simply delete the `pre-commit` file from .git/hooks."
+    );
+
+    let should_install = loop {
+        print!("Would you like to install the git hook?: [y/N] ");
+        io::stdout().flush()?;
+        input.clear();
+        io::stdin().read_line(&mut input)?;
+        break match input.trim().to_lowercase().as_str() {
+            "y" | "yes" => true,
+            "n" | "no" | "" => false,
+            _ => {
+                println!("error: unrecognized option '{}'", input.trim());
+                println!("note: press Ctrl+C to exit");
+                continue;
+            }
+        };
+    };
+
+    Ok(if should_install {
+        let src = src_path.join("src").join("etc").join("pre-commit.sh");
+        let dst = src_path.join(".git").join("hooks").join("pre-commit");
```

Comment:
Small nit: this plays badly with worktrees. Do you mind changing it to use `git rev-parse --git-common-dir` instead?

```
x.py encountered an error -- do you already have the git hook installed?
Not a directory (os error 20)
```
```

Although I guess the user might not want to install it globally for all worktrees ... I'm not sure what the right behavior is here. Let's leave it for now and we can fix it in a follow-up.

Reply:
yeah i also noticed a similar error trying to run a merge commit...i'm kind of starting to get out of my league at this point tho so i'm more than happy to move it to a follow up 😅